source/multigrid/mg_transfer_matrix_free.cc

```diff
@@ -112,7 +117,20 @@ void MGTransferMatrixFree<dim,VectorType>::build
                                                    dirichlet_indices,
                                                    weights_unvectorized,
                                                    this->copy_indices_global_mine,
-                                                   this->ghosted_level_vector);
```

Comment:
> The code generates indices in local MPI space, and creates a vector partitioner that matches this local numbering

but i suppose this partitioner is used to initialize vectors?

> I do not really understand what kind of modification you made here, but that might be somewhere to look.

that's actually quite simple and naïve, i do the opposite of your comment 

> // We use the vectors stored ghosted_level_vector in the base class for
      // keeping ghosted transfer indices. To avoid keeping two very similar
      // vectors, we merge them here.

namely, I introduce an auxiliary `LA::d::Vector` to use in `setup_transfer()` and when it's done, i initialize the actual, possibly block variant of, `this->ghosted_level_vector`.